napari/layers/base/base.py

```diff
@@ -1403,6 +1403,13 @@ def get_ray_intersections(
 
         # create the bounding box in data coordinates
         bounding_box = self._display_bounding_box(dims_displayed)
+        if not world:
```

Comment:
Yeah, that is a good question. The code path I was trying to fix has [world=False hard-coded](https://github.com/napari/napari/blob/1ba738c36d7ac0faf9bc7b5e8a2228cd93e656f2/napari/layers/base/base.py#L1062-L1067). I think I had initially introduced `if False` around the change to minimize potential changes outside of that specific case, but had not gone back and tested for the `world=True` case. In the case with world=True, the shift along axis `i` probably has to also be scaled by the pixel size on that axis. 

Just now, I tried removing the if clause and `test_cursor_ray_3d` and related tests within `test_layers.py` fail due to shifts between the expected results and the output. Probably worth spending a bit of time determining whether we should be applying the shift in these cases and also updating those tests accordingly.

Reply:
I think we don't want the shift in general for layers such as `Points` or `Shapes` that are not derived from `ImageBase`, so perhaps this shift should go somewhere outside of the base Layer class instead.

Reply:
Thanks for the quick follow up, @grlee77 !

> I think we don't want the shift in general for layers such as Points or Shapes that are not derived from ImageBase, so perhaps this shift should go somewhere outside of the base Layer class instead.

I am wondering the same thing. I think this might be better in `ImageBase` or in the `Labels` and `Image` classes.